security/src/main/java/io/micronaut/security/authentication/BasicAuthAuthenticationFetcher.java

```diff
@@ -36,21 +38,21 @@
  * and password from the Authorization header and authenticates the credentials
  * against any {@link AuthenticationProvider}s available.
  */
-@Requires(property = BasicAuthAuthenticationConfiguration.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+@Requires(property = SecurityConfigurationProperties.PREFIX + ".basic-auth.enabled", notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
 @Singleton
 public class BasicAuthAuthenticationFetcher implements AuthenticationFetcher {
 
     private static final Logger LOG = LoggerFactory.getLogger(BasicAuthAuthenticationFetcher.class);
     private static final String PREFIX = HttpHeaderValues.AUTHORIZATION_PREFIX_BASIC + " ";
     private final Authenticator authenticator;
-    private final BasicAuthAuthenticationConfiguration configuration;
+    private final TokenConfiguration configuration;
 
     /**
      * @param authenticator The authenticator to authenticate the credentials
      * @param configuration The basic authentication configuration
      */
     public BasicAuthAuthenticationFetcher(Authenticator authenticator,
-                                          BasicAuthAuthenticationConfiguration configuration) {
+                                          TokenConfiguration configuration) {
```

Comment:
I'm not sure I agree we should be using token configuration here. A basic auth string is not a token, so it doesn't make sense for users to configure it via ...token..

Reply:
Maybe the issue is `TokenConfiguration` should be renamed to `AuthenticationConfiguration`.

Reply:
Please note that the reason for this pr is that not using `TokenConfiguration::getRolesName` cuases a bug. See steps the reproduce here: https://github.com/micronaut-projects/micronaut-security/issues/284